src/main/java/se/liu/ida/hefquin/engine/data/VocabularyMapping.java

```diff
@@ -22,12 +22,14 @@
 	/**
 	 * Applies this vocabulary mapping to the given solution mapping expressed
 	 * in the local vocabulary and returns a resulting set of solution mappings
-	 * expressed in the global vocabulary. 
+	 * expressed in the global vocabulary, if boolean fromLocal is true. Else the
+	 * vocabulary mapping is used to from the global vocabulary to the local
+	 * vocabulary.
 	 * If this mapping is not relevant for the given solution mapping (i.e.,
 	 * applying the mapping to the given solution mapping does not have any
 	 * effect), then the result of this function is a set containing only the
 	 * given solution mapping.
 	 */
-	Set<SolutionMapping> translateSolutionMapping( SolutionMapping sm );
+	Set<SolutionMapping> translateSolutionMapping( SolutionMapping sm , boolean fromLocal);
```

Comment:
I would prefer to have two methods here instead of choosing based on a flag, specially since in the implementation you have a call to two different methods based on this boolean value.